### PR TITLE
:bar_chart: bilateral remittance flows

### PR DESCRIPTION
**Pull request by antea04**
Adding[ bilateral remittance flows from KNOMAD/ WB](https://prosperitydata360.worldbank.org/en/dataset/WB+KNOMAD).

### CODE REVIEW

File: etl/steps/data/garden/wb/2024-12-17/bilateral_remittance.meta.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        description_from_producer: |-
+          The caveats attached to this estimate are:
+          (a) The migrant stock data is drawn from the Bilateral Migration Matrix, which is itself based on UN Population Division and National Census data. These are by nature updated infrequently and may not appropriately capture sudden changes in migrant stock;
+          (b) The incomes of migrants abroad and the costs of living in the migrants' country of origin are both proxied by per capita incomes in PPP terms, which is only a rough proxy;
+          (c) Remittance behavior of second-generation migrants who may be recorded as native-born in the remittance source country are not accounted for;
+          (d) There is no way to capture remittances flowing  through informal, unrecorded channels;
+          (e) It does not account for cases where remittances may be miscalculated due to accounting errors arising from confusion with trade and tourism receipts;
+          (f) It may also include cases of retirees moving to certain countries and taking out (remitting) their life long savings."
+        unit: Current US Dollars
```
Comment (spoonerf) [inline etl/steps/data/garden/wb/2024-12-17/bilateral_remittance.meta.yml:32]: Just to align with the[ harmonizing FAUST](https://docs.google.com/document/d/1tcknTryJAxOR0yv8x38s3jkONq-TY3T4HNCEcZQ_c-s/edit?tab=t.0) doc this should be: 'current US$'

File: etl/steps/data/garden/wb/2024-12-17/bilateral_remittance.meta.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        short_unit: USD
```
Comment (spoonerf) [inline etl/steps/data/garden/wb/2024-12-17/bilateral_remittance.meta.yml:33]: And just '$' here

File: snapshots/wb/2024-12-17/knomad.xlsx.dvc [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    description: |-
+      KNOMAD, the Global Knowledge Partnership on Migration and Development, is a brain trust for the global migration community.
+
+      It is an open, inclusive, multidisciplinary knowledge partnership that draws on experts to create and synthesize knowledge for use by policy makers in sending, receiving and transit countries.
+
+      KNOMAD aims to generate a menu of policy choices, based on analytical evidence, evaluation of policies, data collection, and quality control through peer-review. It provides technical assistance and undertakes pilot projects. KNOMAD works in close coordination with the Global Forum on Migration and Development (GFMD) and the U.N. agencies working on migration.
+    date_published: "2024-03-27"
+    title_snapshot: KNOMAD - Bilateral remittances
+    description_snapshot: |-
+      These estimates are based on the methodology developed by Ratha and Shaw, 2007, "South-South Migration and Remittances," World Bank (https://openknowledge.worldbank.org/handle/10986/6733).
```
Comment (spoonerf) [inline snapshots/wb/2024-12-17/knomad.xlsx.dvc:16]: Maybe a hyperlink here would be nice

File: etl/steps/data/garden/wb/2024-12-17/bilateral_remittance.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+    # Rename and drop columns
+    tb = tb.drop(columns=["indicator_id", "economy_iso3", "indicator"])
+    tb = tb.rename(columns={"value": "remittance_flows"})
+    tb["remittance_flows"] = tb["remittance_flows"].astype(float) * 1e6  # in million USD
+
+    # filter on 2021 as that is the year we have data for
+    tb = tb[tb["year"] == 2021]
+
+    tb.metadata.short_name = "bilateral_remittance"
```
Comment (spoonerf) [inline etl/steps/data/garden/wb/2024-12-17/bilateral_remittance.py:35]: You add this later in format(), so I don't think you need it here?

File: etl/steps/data/garden/wb/2024-12-17/bilateral_remittance.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+    #
+    # Harmonize country names
+    #
+    tb = geo.harmonize_countries(df=tb, country_col="country_origin", countries_file=paths.country_mapping_path)
```
Comment (spoonerf) [inline etl/steps/data/garden/wb/2024-12-17/bilateral_remittance.py:40]: You could set warn_on_unused_countries to False for both of these

File: etl/steps/data/grapher/wb/2024-12-17/bilateral_remittance.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
+"""Load a garden dataset and create a grapher dataset."""
+
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+
+indicator_title = "Estimated bilateral remittance flows"
+description_short = "The estimated bilateral remittance flows between countries. These estimates use migrant stocks, host country incomes, and origin country incomes."
```
Comment (spoonerf) [inline etl/steps/data/grapher/wb/2024-12-17/bilateral_remittance.py:10]: I think aside from indicator_title you can remove these as they persist from garden step

File: etl/steps/data/grapher/wb/2024-12-17/bilateral_remittance.py [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+        tb_cty = tb[tb["country_origin"] == cty].copy()
+        tb_cty = tb_cty.rename(columns={"country_receiving": "country"})
+        tb_cty = tb_cty.drop(columns=["country_origin"])
+
+        col_name = f"remittance_flows_from_{'_'.join([word.lower() for word in cty.split(' ')])}"
+        print(col_name)
+        tb_cty = tb_cty.rename(columns={"remittance_flows": col_name})
+
+        tb_cty[col_name].metadata.title = f"{indicator_title} sent from {cty}"
+        tb_cty[col_name].metadata.short_unit = short_unit
```
Comment (spoonerf) [inline etl/steps/data/grapher/wb/2024-12-17/bilateral_remittance.py:51]: Likewise, apart from title you don't need to define these again

File: etl/steps/data/grapher/wb/2024-12-17/bilateral_remittance.py [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+
+indicator_title = "Estimated bilateral remittance flows"
+description_short = "The estimated bilateral remittance flows between countries. These estimates use migrant stocks, host country incomes, and origin country incomes."
+description_key = [
+    "Flows between India and Pakistan, Pakistan and India, Lebanon and Israel and vice versa, and Azerbaijan and Armenia and vice versa  are assumed to be zero given the political economy situations in these corridors."
+]
+description_from_producer = [
```
Comment (spoonerf) [inline etl/steps/data/grapher/wb/2024-12-17/bilateral_remittance.py:14]: This is also not formatting quite right at the moment, but I think using the garden version should be all good.
Screenshot 2024-12-19 at 10 34 14

**Review by spoonerf — APPROVED** [review]
Looks good, mostly just suggestions around simplifying the metadata,

File: etl/steps/data/garden/wb/2024-12-17/bilateral_remittance.py (hunk 5 above)
Comment (antea04) [reply etl/steps/data/garden/wb/2024-12-17/bilateral_remittance.py:40]: Ah, that's a good idea, thank you!

**Pull request merged**